cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/openai/OpenAIPrompt.scala

```diff
@@ -82,40 +91,75 @@ class OpenAIPrompt(override val uid: String) extends Transformer
   }
 
   private val localParamNames = Seq(
-    "promptTemplate", "outputCol", "postProcessing", "postProcessingOptions", "dropPrompt")
+    "promptTemplate", "outputCol", "postProcessing", "postProcessingOptions", "dropPrompt", "dropMessages")
 
   override def transform(dataset: Dataset[_]): DataFrame = {
     import com.microsoft.azure.synapse.ml.core.schema.DatasetExtensions._
 
     logTransform[DataFrame]({
       val df = dataset.toDF
 
-      val promptColName = df.withDerivativeCol("prompt")
-
-      val dfTemplated = df.withColumn(promptColName, Functions.template(getPromptTemplate))
-
-      val completion = openAICompletion.setPromptCol(promptColName)
-
-      // run completion
-      val results = completion
-        .transform(dfTemplated)
-        .withColumn(getOutputCol,
-          getParser.parse(F.element_at(F.col(completion.getOutputCol).getField("choices"), 1)
-            .getField("text")))
-        .drop(completion.getOutputCol)
-
-      if (getDropPrompt) {
-        results.drop(promptColName)
-      } else {
-        results
+      val completion = openAICompletion
+      val promptCol = Functions.template(getPromptTemplate)
+      val systemPrompt = "You are an AI Chatbot. Only respond with a completion."
```

Comment:
Am using this primitive System Prompt right now - as I'm using the user's prompt as the user message. Let me know if you have other ideas re: this.